src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ISwitchStatement.cs

```diff
@@ -0,0 +1,1213 @@
+    {
+        /*<bind>*/switch (number)
+        {
+            case 0:
+                goto case 1;
+            case 1:
+                break;
+        }/*</bind>*/
+    }
+}
+";
+            string expectedOperationTree = @"
+ISwitchStatement (2 cases) (OperationKind.SwitchStatement) (Syntax: 'switch (num ... }')
+  Switch expression: 
+    IParameterReferenceExpression: number (OperationKind.ParameterReferenceExpression, Type: System.Int32) (Syntax: 'number')
+  Sections:
+      ISwitchCase (1 case clauses, 1 statements) (OperationKind.SwitchCase) (Syntax: 'case 0: ... oto case 1;')
+          Clauses:
+              ISingleValueCaseClause (CaseKind.SingleValue) (OperationKind.CaseClause) (Syntax: 'case 0:')
+                Value: 
+                  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 0) (Syntax: '0')
+          Body:
+              IBranchStatement (BranchKind.GoTo, Label: case 1:) (OperationKind.BranchStatement) (Syntax: 'goto case 1;')
+      ISwitchCase (1 case clauses, 1 statements) (OperationKind.SwitchCase) (Syntax: 'case 1: ... break;')
+          Clauses:
+              ISingleValueCaseClause (CaseKind.SingleValue) (OperationKind.CaseClause) (Syntax: 'case 1:')
+                Value: 
+                  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1) (Syntax: '1')
+          Body:
+              IBranchStatement (BranchKind.Break) (OperationKind.BranchStatement) (Syntax: 'break;')
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<SwitchStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact, WorkItem(17600, "https://github.com/dotnet/roslyn/issues/17600")]
+        public void ISwitchStatement_GoverningExpressionWithLocalDeclarataion()
+        {
+            string source = @"
+using System;
+
+class Test
+{
+    void M()
+    {
+        /*<bind>*/switch (M2(out var number))
+        {
+            case 0:
+                break;
+        }/*</bind>*/
+    }
+
+    int M2(out int number)
+    {
+        number = 0;
+        return 1;
+    }
+}
+";
+            string expectedOperationTree = @"
+ISwitchStatement (1 cases) (OperationKind.SwitchStatement) (Syntax: 'switch (M2( ... }')
+  Switch expression: 
+    IInvocationExpression ( System.Int32 Test.M2(out System.Int32 number)) (OperationKind.InvocationExpression, Type: System.Int32) (Syntax: 'M2(out var number)')
+      Instance Receiver: 
+        IInstanceReferenceExpression (OperationKind.InstanceReferenceExpression, Type: Test, IsImplicit) (Syntax: 'M2')
+      Arguments(1):
+          IArgument (ArgumentKind.Explicit, Matching Parameter: number) (OperationKind.Argument) (Syntax: 'out var number')
+            ILocalReferenceExpression: number (IsDeclaration: True) (OperationKind.LocalReferenceExpression, Type: System.Int32) (Syntax: 'var number')
```

Comment:
>IsDeclaration: True [](start = 47, length = 19)

It would be good to verify that there is a synthetic block with this local in ```Locals``` array.